src/Formatter/PlatesFormatter.php

```diff
@@ -26,27 +25,42 @@ public function __construct(Engine $engine)
      */
     public function body(PayloadInterface $payload)
     {
-        $template = $this->template($payload);
-        $template = $this->engine->make($template);
-        return $this->render($template, $payload);
+        return $this->render($payload);
     }
 
     /**
      * @param PayloadInterface $payload
+     *
      * @return string
      */
+    protected function render(PayloadInterface $payload)
+    {
+        $template = $this->template($payload);
+        $output = $this->output($payload);
+
+        return $this->engine->render($template, $output);
+    }
+
+    /**
+     * @param PayloadInterface $payload
+     *
+     * @return string Template name
+     */
     protected function template(PayloadInterface $payload)
     {
         return $payload->getOutput()['template'];
     }
 
     /**
-     * @param Template $template
      * @param PayloadInterface $payload
-     * @return string
+     *
+     * @return array $output
      */
-    protected function render(Template $template, PayloadInterface $payload)
+    protected function output(PayloadInterface $payload)
     {
-        return $template->render($payload->getOutput());
+        $output = $payload->getOutput();
+        unset($output['template']);
```

Comment:
Probably the "Content Negotiation" is overrated for me. Domain it is still important to know how and in what format it will be delivered data to the user.

Sorry for my stupid English. I will try to explain on examples. 
I wrote about it, the same endpoint, but different client header "Accept":
```bash
$ curl -i -H "Accept:application/json" "http://example.dev"

HTTP/1.1 200 OK
Content-Type: application/json

{"template":"index.php","message":"message"}
```

Just don't ask what this use case, I don't do it. But it could be interesting and cool feature, one endpoint (Domain) and different formatters. Honestly, I don't use the Plates for templates, but wanted to support it. I use the Twig and even made [an integration with Equip](https://github.com/AlexMasterov/equip-twig).